exec/java-exec/src/test/java/org/apache/drill/exec/store/easy/text/compliant/TestCsvWithSchema.java

```diff
@@ -82,6 +167,468 @@ public void testSchema() throws Exception {
+      // because doing so can lead to a flakey test.
+
+      if (!sawFile1First || ! sawFile2First || !sawFullSchema || !sawPartialSchema || !sawIdAsCol0 || !sawIdAsCol1) {
+        System.out.println("Some variations did not occur");
+        System.out.println(String.format("File 1 first: %s", sawFile1First));
+        System.out.println(String.format("File 1 second: %s", sawFile2First));
+        System.out.println(String.format("Full schema: %s", sawFullSchema));
+        System.out.println(String.format("Partial schema: %s", sawPartialSchema));
+        System.out.println(String.format("`id` as col 0: %s", sawIdAsCol0));
+        System.out.println(String.format("`id` as col 1: %s", sawIdAsCol1));
+      }
+      // Sanity checks
+      assertTrue(sawFullSchema);
+      assertTrue(sawFile1First || sawFile2First);
+      assertTrue(sawIdAsCol0 || sawIdAsCol1);
+    } finally {
+      resetV3();
+      resetSchema();
+      resetMultiScan();
+    }
+  }
+
+  /**
+   * Show that, without schema, the hard schema change for the "missing"
+   * gender column causes an error in the sort operator when presented with
+   * one batch in which gender is VARCHAR, another in which it is
+   * Nullable INT. This is a consequence of using SELECT * on a distributed
+   * scan.
+   */
+  @Test
+  public void testWildcardSortFailure() throws Exception {
+    try {
+      enableSchema(false);
+      enableMultiScan();
+      enableV3(false);
+      String tablePath = buildTwoFileTable("wildcardSortV2", multi1Contents, reordered2Contents);
+      doTestWildcardSortFailure(tablePath);
+      enableV3(true);
+      doTestWildcardSortFailure(tablePath);
+    } finally {
+      resetV3();
+      resetSchema();
+      resetMultiScan();
+    }
+  }
+
+  private void doTestWildcardSortFailure(String tablePath) throws Exception {
+    String sql = "SELECT * FROM " + tablePath + " ORDER BY id";
+    boolean sawError = false;
+    for (int i = 0; i < 10; i++) {
+      try {
+        // When this fails it will print a nasty stack trace.
+        RowSet result = client.queryBuilder().sql(sql).rowSet();
+        assertEquals(4, result.rowCount());
+        result.clear();
+      } catch (RpcException e) {
+        assertTrue(e.getCause() instanceof UserRemoteException);
+        sawError = true;
+        break;
+      }
+    }
+    assertTrue(sawError);
+  }
+
+  /**
+   * Test an explicit projection with a sort. Using the sort 1) will blow up
+   * if the internal schema is inconsistent, and 2) allows easier verification
+   * of the merged table results.
+   * <p>
+   * Fails with <code><pre>
+   * #: id, name, gender
+   * 0: "1", "bob  ", "      "
+   * 1: "2", "    ", "    "
+   * 2: "3", " 
+    * ", ""
+   * 3: "4", "  
+   *    " java.lang.NegativeArraySizeException: null
+   *      at io.netty.buffer.DrillBuf.unsafeGetMemory(DrillBuf.java:852) ~[classes/:4.0.48.Final]
+   * </pre></code>
+   */
+  @Test
```

Comment:
Code is not highlighted as code in GitHub, I remember there were similar issues before, if I am not mistaken some hidden symbols were copied from text editor. Could you please check?

Reply:
Looks OK to me. I suspect the problem might be with the volume of changes; that sometimes throws off browsers. Try viewing this again without expanding any other files that are not expanded by default.